src/java/org/apache/cassandra/service/opstate/KeyspaceTableOpStateCache.java

```diff
@@ -63,6 +69,25 @@ public Optional<Instant> getMinimumTsOfAllEntries()
         if (tableEntries.isEmpty())
             return Optional.empty();
 
+        // Remove potentially deleted keyspaces and tables from entries
+        Set<KeyspaceTableKey> invalidEntries =
+            Sets.difference(tableEntries.keySet(), Sets.intersection(tableEntries.keySet(), getValidKeyspaceTableEntries()));
+        invalidEntries.forEach(tableEntries::remove);
+
         return Optional.of(Collections.min(tableEntries.values()));
     }
+
+
+    @VisibleForTesting
+    Set<KeyspaceTableKey> getValidKeyspaceTableEntries(){
+        Set<KeyspaceTableKey> validEntries = new HashSet<>();
+        Schema.instance.getKeyspaces().forEach(keyspace -> {
+            Set<KeyspaceTableKey> keyspaceEntries = Schema.instance.getKeyspaceInstance(keyspace).getColumnFamilyStores()
+                                                                   .stream()
+                                                                   .map(cf -> KeyspaceTableKey.of(keyspace, cf.getColumnFamilyName()))
+                                                                   .collect(Collectors.toSet());
+            validEntries.addAll(keyspaceEntries);
+        });
+        return validEntries;
```

Comment:
nit: can 1-line this

```suggestion
return Schema.instance.getKeyspaces().stream()
                              .map(keyspace -> Schema.instance.getKeyspaceInstance(keyspace).getColumnFamilyStores()
                                                              .stream()
                                                              .map(cf -> KeyspaceTableKey.of(keyspace, cf.getColumnFamilyName()))
                                                              .collect(Collectors.toSet()))
                              .flatMap(Set::stream)
                              .collect(Collectors.toSet());
```

Reply:
I tried using flatMap first but for some reason this converts the stream to a Stream<Object> type which makes the eclipse-warnings fail (see commit above)

Reply:
Hm that's odd. I see [what you're talking about](https://app.circleci.com/pipelines/github/palantir/cassandra/2004/workflows/dfe75481-f000-48d2-ad69-d609888847ad), yet I pushed another branch with your old way and my suggested way here but they both passed 🤔 
https://app.circleci.com/pipelines/github/palantir/cassandra/2012/workflows/f71fb1ed-40f9-49e6-bb6c-e00776f58581

https://app.circleci.com/pipelines/github/palantir/cassandra/2013/workflows/89778490-a4fc-4043-b25d-0c3581d1c816

Reply:
Strange ... I'll push your suggestion then, hopefully that was just a fluke